### PR TITLE
Feature/8 prepare dimensions & props 

### PR DESCRIPTION
**Pull request by jasrial**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,72 @@
...
     <br />
     <br />
-    <Ruleset />
+    <TagRules rulesets={rulesets} dimensions={dimensions} />
+    <br />
+    <br />
   </div>
 );
 
+const incomingDimensions = [
```
Comment (vlabs) [inline src/App.js]: move to a separate file `src/mocks/dimensions.js`

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,72 @@
...
+  { id: "reporting_campaign", name: "Reporting Campaign" },
+  { id: "section", name: "Section" },
+  { id: "source_campaign", name: "Source Campaign" },
+  { id: "targeting", name: "Targeting" },
+  { id: "unit", name: "Unit" }
+];
+
+const dimensions = prepareDimensions(incomingDimensions);
+
+const rulesets = [
```
Comment (vlabs) [inline src/App.js]: move to a separate file `src/mocks/ruleSets.js`

imho two words formatting fits better: rule**S**et

File: src/App.js [hunk 3]
```diff
@@ -1,12 +1,72 @@
...
+  { id: "objective", name: "Objective" },
+  { id: "publisher", name: "Publisher" },
+  { id: "reporting_campaign", name: "Reporting Campaign" },
+  { id: "section", name: "Section" },
+  { id: "source_campaign", name: "Source Campaign" },
+  { id: "targeting", name: "Targeting" },
+  { id: "unit", name: "Unit" }
+];
+
+const dimensions = prepareDimensions(incomingDimensions);
```
Comment (vlabs) [inline src/App.js]: should be done above `App`

File: src/components/TagRules.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faTrash } from "@fortawesome/free-solid-svg-icons";
+import { Container, Row, Col, Button } from "react-bootstrap";
+import "./TagRules.css";
+import PropTypes from "prop-types";
+
+const TagRules = ({ rulesets, onChange, dimensions }) => (
+  <div>
+    <Container fluid>
+      {rulesets.map(item => (
```
Comment (vlabs) [inline src/components/TagRules.js]: name `item` => `ruleSet`

File: src/components/Ruleset.js [hunk 5]
```diff
@@ -5,36 +5,43 @@ import Select from "react-select";
 import "./Ruleset.css";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faTrash } from "@fortawesome/free-solid-svg-icons";
+import PropTypes from "prop-types";
 
-const Ruleset = props => (
+const Ruleset = ({ ruleset, onChange, dimensions }) => (
```
Comment (vlabs) [inline src/components/Ruleset.js]: pass `rules` and `dimension` explicitly

File: src/components/TagRules.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    </Container>
+  </div>
+);
+
+TagRules.propTypes = {
+  rulesets: PropTypes.arrayOf(
+    PropTypes.shape({
+      sourceDimension: PropTypes.string,
+      rules: PropTypes.arrayOf(
+        PropTypes.shape({
```
Comment (vlabs) [inline src/components/TagRules.js]: this shape has been declared three times, try to reuse it

File: src/App.js (hunk 1 above)
Comment (jasrial) [reply src/App.js]: 9bf2ea7b8288c9a35c965d5a40728f1c8a395375

File: src/App.js (hunk 2 above)
Comment (jasrial) [reply src/App.js]: d43178851e3d988928a6271d57b24203d33b75ea

File: src/App.js (hunk 3 above)
Comment (jasrial) [reply src/App.js]: 8fc19a9a87f05bbc3b14ecc11d245744d5643d5b

File: src/components/TagRules.js (hunk 4 above)
Comment (jasrial) [reply src/components/TagRules.js]: a07e4b9c27e48e95da72504567686b93708e5d13

File: src/components/Ruleset.js (hunk 5 above)
Comment (jasrial) [reply src/components/Ruleset.js]: 2626124e99fc76e865f07919ee29ba26a84288bf

Comment (jasrial) [reply src/components/Ruleset.js]: e7242075e0b829caf27fb12a9bc677b7141fb5a7

File: src/components/Rule.js [hunk 7]
```diff
@@ -3,44 +3,71 @@ import { Container, Row, Col, FormControl } from "react-bootstrap";
...
         </Col>
         <Col md={4}>
-          <FormControl />
+          <FormControl value={rule.tag} />
         </Col>
       </Row>
       <Row className="Row">
         <Col md={3}>
-          <p style={{ fontWeight: "bold" }}>Into target dimension:</p>
+          <p class="font-weight-bold">Into target dimensions:</p>
```
Comment (vlabs) [inline src/components/Rule.js]: singular!

File: src/components/Rule.js [hunk 8]
```diff
@@ -3,44 +3,71 @@ import { Container, Row, Col, FormControl } from "react-bootstrap";
...
         </Col>
         <Col md={4}>
-          <Select placeholder="Choose dimension..." />
+          <Select
+            placeholder="Choose dimensions..."
```
Comment (vlabs) [inline src/components/Rule.js]: singular!

File: src/components/Rule.js [hunk 9]
```diff
@@ -3,44 +3,71 @@ import { Container, Row, Col, FormControl } from "react-bootstrap";
...
+            options={dimensions}
+            value={{ value: rule.tagType, label: rule.tagType }}
+          />
         </Col>
         <Col md={4}>
-          <FormControl placeholder="Or enter new dimension..." />
+          <FormControl
+            placeholder="Or enter new dimensions..."
```
Comment (vlabs) [inline src/components/Rule.js]: singular!

File: src/components/RuleSet.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+  <div className="RuleSet">
+    <Container fluid>
+      <Row className="SetRow">
+        <Col md={3}>
+          <p class="font-weight-bold">For source dimensions:</p>
+        </Col>
+        <Col md={5}>
+          <Select
+            options={dimensions}
+            value={{
```
Comment (vlabs) [inline src/components/RuleSet.js]: shouldn't this be just `value={dimension}`

File: src/components/TagRules.js (hunk 6 above)
Comment (jasrial) [reply src/components/TagRules.js]: 76d3e98f1dff914b5ac4d512ca11c04bde0efc17

File: src/components/Rule.js (hunk 9 above)
Comment (jasrial) [reply src/components/Rule.js]: 5c8c8ef4083438cb541f7b555d2fdd82a777dd8d

File: src/components/Rule.js (hunk 8 above)
Comment (jasrial) [reply src/components/Rule.js]: 5c8c8ef4083438cb541f7b555d2fdd82a777dd8d

File: src/components/Rule.js (hunk 7 above)
Comment (jasrial) [reply src/components/Rule.js]: 5c8c8ef4083438cb541f7b555d2fdd82a777dd8d

File: src/components/RuleSet.js (hunk 10 above)
Comment (jasrial) [reply src/components/RuleSet.js]: 9c2c6769a6105140ed55b1bbb1e6537374cc4590

File: src/components/Rule.js [hunk 11]
```diff
@@ -3,44 +3,70 @@ import { Container, Row, Col, FormControl } from "react-bootstrap";
...
         <Col md={3}>
-          <p style={{ fontWeight: "bold" }}>Into target dimension:</p>
+          <p class="font-weight-bold">Into target dimension:</p>
         </Col>
         <Col md={4}>
-          <Select placeholder="Choose dimension..." />
+          <Select
+            placeholder="Choose dimension..."
+            options={dimensions}
+            value={dimensions.find(element => element.value === rule.tagType)}
```
Comment (vlabs) [inline src/components/Rule.js]: :+1:

File: src/components/Rule.js [hunk 12]
```diff
@@ -3,44 +3,70 @@ import { Container, Row, Col, FormControl } from "react-bootstrap";
...
         </Col>
         <Col md={4}>
-          <FormControl placeholder="Or enter new dimension..." />
+          <FormControl
+            placeholder="Or enter new dimension..."
+            disabled={dimensions.find(
+              element => element.value === rule.tagType
+            )}
+            value={
+              dimensions.find(element => element.value === rule.tagType)
```
Comment (vlabs) [inline src/components/Rule.js]: now it's the third time (DRY)
please extract to util function

File: src/components/RuleSet.js [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  <div className="RuleSet">
+    <Container fluid>
+      <Row className="SetRow">
+        <Col md={3}>
+          <p class="font-weight-bold">For source dimension:</p>
+        </Col>
+        <Col md={5}>
+          <Select
+            options={dimensions}
+            value={dimensions.find(element => element.value === dimension)}
```
Comment (vlabs) [inline src/components/RuleSet.js]: here too :)

File: src/components/Rule.js (hunk 12 above)
Comment (jasrial) [reply src/components/Rule.js]: 874519e6988ac08c7b6b56ef2e8a2e7c96c10698

File: src/components/RuleSet.js (hunk 13 above)
Comment (jasrial) [reply src/components/RuleSet.js]: 874519e6988ac08c7b6b56ef2e8a2e7c96c10698

**Pull request closed without merging**